spaceapi/src/lib.rs

```diff
@@ -8,11 +8,35 @@ use rustc_serialize::json::{Json, ToJson};
 /// An Optional value can contain Optional::Some<T> or Optional::Absent.
 /// It is similar to an Option, but Optional::Absent means it will be
 /// omitted when serialized.
+#[derive(Debug, Copy, Clone)]
 pub enum Optional<T> {
     Value(T),
     Absent,
 }
 
+impl<T> Optional<T> {
+    pub fn map_or<U, F: FnOnce(T) -> U>(self, def: U, f: F) -> U {
+        match self {
+            Optional::Value(v) => f(v),
+            Optional::Absent => def,
+        }
+    }
```

Comment:
Ok. Not necessary to add a doc-string. It's ok, since the name has already been chosen by the rust people.

But still:
Why are we using `map_or`? `ToJson` is implemented for `Option<A>`. So if we'd use `Option` instead of `Optional`, we could save a lot of code.